zstd is compressing symbolic links with `std=c99`

@Cyan4973 Hmm, I might find a test error that zstd have been 
compressed the symbolic link.

_Originally posted by @lzutao in https://github.com/facebook/zstd/issues/1605#issuecomment-487978531_

This may not be relevant but I get this warning when building
with `make c99build` on CentOS 5 (gcc (GCC) 5.5.0)

```c
gcc -I../lib -I../lib/common -I../lib/compress -I../lib/dictBuilder -I../lib/deprecated -I../programs -std=c99 -Werror -Wall -Wextra -Wcast-qual -Wcast-align -Wshadow -Wstrict-aliasing=1 -Wswitch-enum -Wdeclaration-after-statement -Wstrict-prototypes -Wundef -Wvla -Wformat=2 -Winit-self -Wfloat-equal -Wwrite-strings -Wredundant-decls -Wmissing-prototypes -g -DDEBUGLEVEL=1   -DZSTD_MULTITHREAD -pthread ../programs/util.c ../programs/timefn.c poolTests.c ../lib/common/pool.c ../lib/common/threading.c ../lib/common/zstd_common.c ../lib/common/error_private.c -o poolTests
In file included from poolTests.c:14:0:
poolTests.c: In function 'waitFn':
../programs/util.h:66:99: error: implicit declaration of function 'nanosleep' [-Werror=implicit-function-declaration]
 #      define UTIL_sleepMilli(milli) { struct timespec t; t.tv_sec=0; t.tv_nsec=milli*1000000ULL; nanosleep(&t, NULL); }
                                                                                                   ^
poolTests.c:71:3: note: in expansion of macro 'UTIL_sleepMilli'
   UTIL_sleepMilli(1);
   ^
cc1: all warnings being treated as errors
```

So `zstd` is being tested with `std=gnu99` by default?
I think I should change to the same `std` version in meson 
build file to avoid error when using run `meson test`.

--- a/.travis.yml
+++ b/.travis.yml
@@ -153,6 +153,12 @@ matrix:
         - make clean
         - make travis-install    # just ensures `make install` works
 
+    - name: c99 compatibility
+      if: branch = master
+      script:
+        - make c99build
+        - make -C tests test-zstd
+
     - name: PPC64
       if: branch = master
       script:

--- a/programs/util.c
+++ b/programs/util.c
@@ -107,19 +107,11 @@ int UTIL_isSameFile(const char* file1, const char* file2)
 U32 UTIL_isLink(const char* infilename)
 {
 /* macro guards, as defined in : https://linux.die.net/man/2/lstat */
-#ifndef __STRICT_ANSI__
-#if defined(_BSD_SOURCE) \
-    || (defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)) \
-    || (defined(_XOPEN_SOURCE) && defined(_XOPEN_SOURCE_EXTENDED)) \
-    || (defined(_POSIX_C_SOURCE) && (_POSIX_C_SOURCE >= 200112L)) \
-    || (defined(__APPLE__) && defined(__MACH__)) \
-    || defined(__OpenBSD__) \
-    || defined(__FreeBSD__)
+#if PLATFORM_POSIX_VERSION >= 200112L
     int r;
     stat_t statbuf;
     r = lstat(infilename, &statbuf);
     if (!r && S_ISLNK(statbuf.st_mode)) return 1;
-#endif
 #endif
     (void)infilename;
     return 0;

--- a/tests/Makefile
+++ b/tests/Makefile
@@ -247,7 +247,7 @@ clean:
 	$(MAKE) -C $(ZSTDDIR) clean
 	$(MAKE) -C $(PRGDIR) clean
 	@$(RM) -fR $(TESTARTEFACT)
-	@$(RM) -f core *.o tmp* result* *.gcda dictionary *.zst \
+	@$(RM) -f core *.o tmp* *.tmp result* *.gcda dictionary *.zst \
         $(PRGDIR)/zstd$(EXT) $(PRGDIR)/zstd32$(EXT) \
         fullbench$(EXT) fullbench32$(EXT) \
         fullbench-lib$(EXT) fullbench-dll$(EXT) \

